### PR TITLE
Upgrade Tasty to 1.0 and update appveyor.yml.

### PR DESCRIPTION
**Pull request by phadej**
Also set an upper bound on Tasty, because it tends to break things
when updated.

Also require optparse-applicative >= 0.14 in cabal-testsuite

Supersedes #5014  #5016 

---

Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by phadej** [comment]
I thought there was revision for `hackage-repo-tool` already :/

**Comment by 23Skidoo** [comment]
> I thought there was revision for `hackage-repo-tool` already :/

Is only a Hackage revision required or a new package release?

**Comment by phadej** [comment]
I'll just relax it via `allow-newer`.

File: cabal.project [hunk 1]
```diff
@@ -5,6 +5,10 @@ constraints: unix >= 2.7.1.0
 -- Uncomment to allow picking up extra local unpacked deps:
 --optional-packages: */
 
+-- Remove after hackage-repo-tool release
```
Comment (23Skidoo) [inline cabal.project]: If this works, I suggest we just make a Hackage revision for `hackage-repo-tool` instead of updating `appveyor.yml` as well.

Comment (phadej) [reply cabal.project]: appveyor doesn't use hackage-repo-tool yet :)

Comment (23Skidoo) [reply cabal.project]: Oh right. Anyway, I just checked that `cabal new-build --allow-newer=optparse-applicative --constraint="optparse-applicative == 0.14.0.0"` works with `hackage-repo-tool-0.1.1`, so I asked @hvr to bump that constraint to `< 0.15`.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Comment by 23Skidoo** [comment]
Also cherry-picked into `master`.

File: cabal.project (hunk 1 above)
Comment (23Skidoo) [reply cabal.project]: Apparently he wants to wait until a new release, which will also include the Windows patch.

**Pull request merged**